art/preprocessing/standardisation_mean_std/standardisation_mean_std.py

```diff
@@ -66,18 +70,24 @@ def __call__(self, x: np.ndarray, y: Optional[np.ndarray] = None,) -> Tuple[np.n
                 "np.float32.".format(x.dtype)
             )
 
-        mean = np.asarray(self.mean, dtype=ART_NUMPY_DTYPE)
-        std = np.asarray(self.std, dtype=ART_NUMPY_DTYPE)
-
+        mean, std = broadcastable_mean_std(x, self.mean, self.std)
         x_norm = x - mean
         x_norm = x_norm / std
         x_norm = x_norm.astype(ART_NUMPY_DTYPE)
 
         return x_norm, y
 
     def estimate_gradient(self, x: np.ndarray, gradient: np.ndarray) -> np.ndarray:
+        """
+        Provide an estimate of the gradients of preprocessor for the backward pass. If the preprocessor is not
+        differentiable, this is an estimate of the gradient, most often replacing the computation performed by the
+        preprocessor with the identity function (the default).
 
-        std = np.asarray(self.std, dtype=ART_NUMPY_DTYPE)
+        :param x: Input data for which the gradient is estimated. First dimension is the batch size.
+        :param grad: Gradient value so far.
+        :return: The gradient (estimate) of the defence.
+        """
+        _, std = broadcastable_mean_std(x, self.mean, self.std)
```

Comment:
Would it be possible to run `broadcastable_mean_std` outside of method `estimate_gradient`? It should only be required once.

Reply:
See above.